src/index.js

```diff
@@ -248,14 +248,21 @@ function EvolvClient(options) {
   /**
    * Marks a consumer as unsuccessfully retrieving and / or applying requested values, making them ineligible for
    * inclusion in optimization statistics.
+   *
+   * @param details {Object} Optional. Information on the reason for contamination. If provided, the object should
+   * contain a reason ('timeout-exceeded',  'error-thrown' or 'user-generated') and optional details
    */
-  this.contaminate = function() {
+  this.contaminate = function(details) {
     const remoteContext = context.remoteContext;
     const allocations = (remoteContext.experiments || {}).allocations // undefined is a valid state, we want to know if its undefined
     if (!allocations || !allocations.length) {
       return;
     }
 
+    if (details && !details.reason) {
```

Comment:
I'm not following why `details.reason` is a distinct and required property. The docstring lists 3 reason types but then later on in the test you use "broken". Is `reason` supposed to be consumed by some non-human agent?

Reply:
The doc string is wrong -- that's from a previous version where I was going to restrict the reason strings.

I'm just enforcing that 'reason' must be sent, so we have consistent field to search for